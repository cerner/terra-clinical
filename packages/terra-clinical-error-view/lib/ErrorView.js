'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _IconError = require('terra-icon/lib/icon/IconError');

var _IconError2 = _interopRequireDefault(_IconError);

var _terraButton = require('terra-button');

var _terraButton2 = _interopRequireDefault(_terraButton);

require('./ErrorView.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /**
   * The name of the error to display.
   */
  name: _react.PropTypes.string,
  /**
   * The error description to display.
   */
  description: _react.PropTypes.string,
  /**
   * The display text for the button.
   */
  buttonText: _react.PropTypes.string,
  /**
   * The click callback function that should be applied to the button.
   */
  buttonOnClick: _react.PropTypes.func,
  /**
   * Whether or not the error glyph should be displayed.
   */
  isGlyphHidden: _react.PropTypes.bool
};

var defaultProps = {
  name: '',
  description: '',
  buttonText: '',
  buttonOnClick: undefined,
  isGlyphHidden: false
};

var ErrorView = function ErrorView(_ref) {
  var name = _ref.name,
      description = _ref.description,
      buttonText = _ref.buttonText,
      buttonOnClick = _ref.buttonOnClick,
      isGlyphHidden = _ref.isGlyphHidden,
      customProps = _objectWithoutProperties(_ref, ['name', 'description', 'buttonText', 'buttonOnClick', 'isGlyphHidden']);

  var errorClassNames = (0, _classnames2.default)(['terraClinical-ErrorView', customProps.className]);

  var glyphSection = void 0;
  if (!isGlyphHidden) {
    var errorIcon = _react2.default.createElement(_IconError2.default, { height: '170', width: '170' });
    glyphSection = _react2.default.createElement(
      'div',
      { className: 'terraClinical-ErrorView-glyph' },
      errorIcon
    );
  }

  var nameSection = void 0;
  if (name) {
    nameSection = _react2.default.createElement(
      'b',
      { className: 'terraClinical-ErrorView-name' },
      name
    );
  }

  var buttonSection = void 0;
  if (buttonText) {
    var button = _react2.default.createElement(_terraButton2.default, { variant: 'secondary', text: buttonText, onClick: buttonOnClick });
    buttonSection = _react2.default.createElement(
      'div',
      { className: 'terraClinical-ErrorView-button' },
      button
    );
  }

  return _react2.default.createElement(
    'div',
    _extends({}, customProps, { className: errorClassNames }),
    glyphSection,
    _react2.default.createElement(
      'p',
      { className: 'terraClinical-ErrorView-text' },
      nameSection,
      description
    ),
    buttonSection
  );
};

ErrorView.propTypes = propTypes;
ErrorView.defaultProps = defaultProps;

exports.default = ErrorView;