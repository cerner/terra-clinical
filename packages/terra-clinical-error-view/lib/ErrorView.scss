@import '~terra-mixins';

.terraClinical-ErrorView {
  align-items: center;
  display: flex;
  flex-direction: column;
  padding: 2.857rem 0 1.429rem;
  width: 100%;
}

.terraClinical-ErrorView-glyph {
  flex: 0 0 auto;
  padding-bottom: 0.2857rem;
}

.terraClinical-ErrorView-text {
  @include terra-text-align-center();
  color: $terra-grey-25;
  flex: 0 0 auto;
  font-size: 1.7142857142857142rem;
  line-height: 1.284;
}

.terraClinical-ErrorView-button {
  flex: 0 0 auto;
  padding-top: 0.2857rem;
}
