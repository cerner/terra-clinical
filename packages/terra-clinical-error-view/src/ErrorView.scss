/* stylelint-disable selector-class-pattern  */
:local {
  .error-view {
    align-items: center;
    display: flex;
    flex-direction: column;
    padding: 1.429rem 0;
    width: 100%;
  }

  .glyph {
    flex: 0 0 auto;
    padding-bottom: 0.2857rem;
    padding-top: 1.429rem;
  }

  .text {
    color: #c8cacb;
    flex: 0 0 auto;
    font-size: 1.7142857142857142rem;
    line-height: 1.284;
    margin-bottom: 0;
    margin-top: 0;
    text-align: center;
  }

  .button {
    flex: 0 0 auto;
    padding-top: 0.2857rem;
  }
}
