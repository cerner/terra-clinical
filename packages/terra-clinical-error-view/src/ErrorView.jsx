import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import Button from 'terra-button';
import ErrorIcon from './ErrorIcon';
import styles from './ErrorView.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The name of the error to display.
   */
  name: PropTypes.string,
  /**
   * The error description to display.
   */
  description: PropTypes.string,
  /**
   * The display text for the button.
   */
  buttonText: PropTypes.string,
  /**
   * The click callback function that should be applied to the button.
   */
  buttonOnClick: PropTypes.func,
  /**
   * Whether or not the error glyph should be displayed.
   */
  isGlyphHidden: PropTypes.bool,
};

const defaultProps = {
  name: '',
  description: '',
  buttonText: '',
  buttonOnClick: undefined,
  isGlyphHidden: false,
};

const ErrorView = ({
    name,
    description,
    buttonText,
    buttonOnClick,
    isGlyphHidden,
    ...customProps
  }) => {
  let glyphSection;
  if (!isGlyphHidden) {
    const errorIcon = <ErrorIcon />;
    glyphSection = <div className={cx('glyph')}>{errorIcon}</div>;
  }

  let nameSection;
  if (name) {
    let nameDisplay = name;
    if (name && description) {
      nameDisplay += '.  ';
    }
    nameSection = <b>{nameDisplay}</b>;
  }

  let buttonSection;
  if (buttonText) {
    const button = <Button variant="secondary" text={buttonText} onClick={buttonOnClick} />;
    buttonSection = <div className={cx('button')}>{button}</div>;
  }

  return (
    <div {...customProps} className={cx('error-view', customProps.className)}>
      {glyphSection}
      <p className={cx('text')}>
        {nameSection}
        {description}
      </p>
      {buttonSection}
    </div>
  );
};

ErrorView.propTypes = propTypes;
ErrorView.defaultProps = defaultProps;

export default ErrorView;
