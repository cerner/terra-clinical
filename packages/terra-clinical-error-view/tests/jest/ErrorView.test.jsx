import React from 'react';
import ErrorView from '../../src/ErrorView';

const defaultVariety = <ErrorView />;

// Snapshot Tests
it('should render a default component', () => {
  const wrapper = render(defaultVariety);
  expect(wrapper).toMatchSnapshot();
});


// Prop Tests
it('should have the class terraClinical-ErrorView', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('error-view');
});

it('should render a name', () => {
  const errorView = <ErrorView name="test error name:" />;
  const wrapper = render(errorView);
  expect(wrapper).toMatchSnapshot();
});

it('should render a description', () => {
  const errorView = <ErrorView description="test description." />;
  const wrapper = render(errorView);
  expect(wrapper).toMatchSnapshot();
});

it('should render a button with text', () => {
  const clickEvent = jest.fn();
  const errorView = <ErrorView buttonText="test button text" buttonOnClick={clickEvent} />;
  const wrapper = shallow(errorView);
  expect(wrapper).toMatchSnapshot();
});

it('should not render a glyph', () => {
  const errorView = <ErrorView isGlyphHidden />;
  const wrapper = render(errorView);
  expect(wrapper).toMatchSnapshot();
});
