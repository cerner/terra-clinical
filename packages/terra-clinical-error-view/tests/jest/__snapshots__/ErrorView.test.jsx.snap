exports[`test should not render a glyph 1`] = `
<div
  class="error-view">
  <p
    class="text" />
</div>
`;

exports[`test should render a button with text 1`] = `
<div
  className="error-view">
  <div
    className="glyph">
    <ErrorIcon />
  </div>
  <p
    className="text" />
  <div
    className="button">
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isReversed={false}
      onClick={[Function]}
      text="test button text"
      type="button"
      variant="secondary" />
  </div>
</div>
`;

exports[`test should render a default component 1`] = `
<div
  class="error-view">
  <div
    class="glyph">
    <svg
      data-name="Layer 1"
      height="170px"
      id="Layer_1"
      viewbox="0 0 251.7 251.8"
      width="170px">
      <path
        d="M125.9,0C56.1,0,0,56.2,0,125.9s56,125.9,125.8,125.9s125.9-56.1,125.9-125.9S195.6,0,125.9,0z M36.7,139 v-26.2H215V139H36.7z"
        style="fill:#e8e9ea;" />
    </svg>
  </div>
  <p
    class="text" />
</div>
`;

exports[`test should render a description 1`] = `
<div
  class="error-view">
  <div
    class="glyph">
    <svg
      data-name="Layer 1"
      height="170px"
      id="Layer_1"
      viewbox="0 0 251.7 251.8"
      width="170px">
      <path
        d="M125.9,0C56.1,0,0,56.2,0,125.9s56,125.9,125.8,125.9s125.9-56.1,125.9-125.9S195.6,0,125.9,0z M36.7,139 v-26.2H215V139H36.7z"
        style="fill:#e8e9ea;" />
    </svg>
  </div>
  <p
    class="text">
    test description.
  </p>
</div>
`;

exports[`test should render a name 1`] = `
<div
  class="error-view">
  <div
    class="glyph">
    <svg
      data-name="Layer 1"
      height="170px"
      id="Layer_1"
      viewbox="0 0 251.7 251.8"
      width="170px">
      <path
        d="M125.9,0C56.1,0,0,56.2,0,125.9s56,125.9,125.8,125.9s125.9-56.1,125.9-125.9S195.6,0,125.9,0z M36.7,139 v-26.2H215V139H36.7z"
        style="fill:#e8e9ea;" />
    </svg>
  </div>
  <p
    class="text">
    <b>
      test error name:
    </b>
  </p>
</div>
`;
